src/main/java/edu/fatec/Porygon/service/DataScrapperService.java

```diff
@@ -79,34 +77,42 @@ public void scrapeDatabyPortalID(int id){
                 noticia.setData(dataPublicacao);
                 noticia.setPortal(portal);
                 noticia.setConteudo(contentScrapper);
+                noticia.setHref(link); // Adicione isso para armazenar o href
 
-                noticias.add(noticia);
+                // Verifica se a notícia já existe no banco
+                if (!noticiaRepository.existsByHref(noticia.getHref())) {
+                    noticias.add(noticia); // Adiciona à lista se não existir
+                } else {
+                    System.out.println("Notícia já existente: " + noticia.getHref());
```

Comment:
No meu teste, deu certo no terminal a verificação. Realmente não foram salvas as notícias repetidas.